packages/flame/lib/src/components/base_component.dart

```diff
@@ -4,30 +4,16 @@ import 'package:meta/meta.dart';
 
 import '../../game.dart';
 import '../../input.dart';
-import '../effects/effects.dart';
-import '../effects/effects_handler.dart';
 import '../extensions/vector2.dart';
 import '../text.dart';
 import 'component.dart';
-import 'component_set.dart';
-import 'mixins/has_game_ref.dart';
 
 /// This can be extended to represent a basic Component for your game.
 ///
-/// The difference between this and [Component] is that the [BaseComponent] can
-/// have children, handle effects and can be used to see whether a position on
-/// the screen is on your component, which is useful for handling gestures.
-abstract class BaseComponent extends Component {
-  final EffectsHandler _effectsHandler = EffectsHandler();
-
-  late final ComponentSet children = createComponentSet();
-
-  /// If the component has a parent it will be set here
-  BaseComponent? _parent;
-
-  @override
-  BaseComponent? get parent => _parent;
-
+/// The difference between this and [Component] is that the [BaseComponent]
+/// implements the [update] and [render] methods, meanwhile you have to do that
```

Comment:
implements the [...] methods => adds a default implement for the [...] methods that passes along the calls to its children